lab5/utils/CommandManager.java

```diff
@@ -0,0 +1,106 @@
+package org.example.utils;
+
+import org.example.commands.*;
+
+import java.util.ArrayDeque;
+
+public class CommandManager {
+    private final CollectionManager collectionManager;
+    private final ArrayDeque<String> commandHistory = new ArrayDeque<>();
+
+    public CommandManager(CollectionManager collectionManager) {
+        this.collectionManager = collectionManager;
+    }
+
+    public void executeCommand(String command) {
+        String[] commandArray = command.split(" ");
+        String c = commandArray[0];
+        String arg = "";
+
+        if(commandArray.length == 2){
+            arg = commandArray[1];
+        }
+
+        switch (c) {
+            case "show":
+                new Show(collectionManager).execute();
+                addToHistory(c);
+                break;
+            case "help":
+                new Help().execute();
+                addToHistory(c);
+                break;
+            case "info":
+                new Info(collectionManager).execute();
+                addToHistory(c);
+                break;
+            case "exit":
+                new Exit().execute();
+                addToHistory(c);
+                break;
+            case "save":
+                new Save(collectionManager).execute();
+                addToHistory(c);
+                break;
+            case "add":
+                new Add(collectionManager).execute();
+                addToHistory(c);
+                break;
+            case "clear":
+                new Clear(collectionManager).execute();
+                addToHistory(c);
+                break;
+            case "history":
+                new History(commandHistory).execute();
+                addToHistory(c);
```

Comment:
why is addToHistory in the switch case?
you can just put it at the end so it will be executed for every case